cmd/govim/buffer_events.go

```diff
@@ -151,6 +172,17 @@ func (v *vimstate) deleteCurrentBuffer(args ...json.RawMessage) error {
 	if !ok {
 		return fmt.Errorf("tried to remove buffer %v; but we have no record of it", currBufNr)
 	}
+
+	// The diagnosticsCache is updated with -1 (unknown buffer) as bufnr.
+	// We don't want to remove the entries completely here since we want to show them in
+	// the quickfix window. And we don't need to remove existing signs or text properties
+	// either here since they are removed by vim automatically when a buffer is deleted.
+	for i := range v.diagnosticsCache {
```

Comment:
This then becomes a lookup by absolute filepath of the buffer (which we have in hand)

Reply:
Good point